docs/configs-api.md

```diff
@@ -0,0 +1,114 @@
+# Configs API
+
+The configs service provides an API-driven multi-tenant approach to handling various configuration files for prometheus. The service hosts an API where users can read and write Prometheus rule files, Alertmanager configuration files, and Alertmanager templates to a database.
+
+Each tenant will have it's own set of rule files, Alertmanager config, and templates. A POST operation will effectively replace the existing copy with the configs provided in the request body.
+
+## Configs Format
+
+At the current time of writing, the API is part-way through a migration from a single Configs service that handled all three sets of data to a split API ([Tracking issue](https://github.com/cortexproject/cortex/issues/619)). All APIs take and return all sets of data.
+
+The following schema is used both when retrieving the current configs from the API and when setting new configs via the API.
+
+### Schema:
+
+```json
+{
+    "id": 99,
+    "rule_format_version": "2",
+    "config": {
+        "alertmanager_config": "<standard alertmanager.yaml config>",
+        "rules_files": { 
+            "rules.yaml": "<standard rules.yaml config>",
+            "rules2.yaml": "<standard rules.yaml config>"
+         },
+        "template_files": {
+            "templates.tmpl": "<standard template file>",
+            "templates2.tmpl": "<standard template file>"
+        }
+    }
+}
+```
+
+### Formatting
+
+`id` - should be incremented every time data is updated; Cortex will use the config with the highest number.
+
+`rule_format_version` - allows compatability for tenants with config in Prometheus V1 format. Pass "1" or "2" according to which Prometheus version you want to match.
```

Comment:
"compatability" is triggering a CI failure - better fix that or CI will be broken on master